test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -88,3 +94,186 @@ def test_invalid_script_out(self):
+                                       body=script_body_s)
+        (fd_m, fn_m) = self.du.mkstemp(prefix="PtlPbs_medium", suffix=".scr",
+                                       body=script_body_m)
+        (fd_l, fn_l) = self.du.mkstemp(prefix="PtlPbs_large", suffix=".scr",
+                                       body=script_body_l)
+
+        self.du.chmod(path=fn_s, mode=0755, sudo=True)
+        self.du.chmod(path=fn_m, mode=0755, sudo=True)
+        self.du.chmod(path=fn_l, mode=0755, sudo=True)
+
+        # Close file handles, else scheduler cannot execute them
+        os.close(fd_s)
+        os.close(fd_m)
+        os.close(fd_l)
+
+        resval = ['"foobar_small !' + fn_s + '"',
+                  '"foobar_medium !' + fn_m + '"',
+                  '"foobar_large !' + fn_l + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        a = {'Resource_List.foobar_small': '4'}
+        # Submit job
+        jid = self.submit_job(a)
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foobar_small': 4}
+        self.server.expect(JOB, a, id=jid)
+
+        a = {'Resource_List.foobar_medium': '10'}
+        # Submit job
+        jid = self.submit_job(a)
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foobar_medium': 10}
+        self.server.expect(JOB, a, id=jid)
+
+        a = {'Resource_List.foobar_large': '18'}
+        # Submit job
+        jid = self.submit_job(a)
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foobar_large': 18}
+        self.server.expect(JOB, a, id=jid)
+
+    # Helper function to setup server dynamic resources
+    def setup_dyn_res(self, resname, restype, resval):
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        attr = {"type": restype}
+        for i in resname:
+            self.server.manager(MGR_CMD_CREATE, RSC, attr, id=i, expect=True)
+            # Add resource to sched_config's 'resources' line
+            self.scheduler.add_resource(i)
+
+        # Add server_dyn_res entry in sched_config
+        if len(resval) > 1:  # Mutliple resources
+            # To create multiple server dynamic resources in sched_config
+            # from PTL, a list containing "resource !<script>" should be
+            # supplied as value to the key 'server_dyn_res' when calling
+            # set_sched_config().
+            # But this workaround works only if sched_config already has a
+            # server_dyn_res entry.
+            # HACK: So adding a single resource first and then the list.
+            # There wouldn't be any duplicate entries though.
+            a = {'server_dyn_res': resval[0]}
+            self.scheduler.set_sched_config(a)
+            a = {'server_dyn_res': resval}
+        else:
+            a = {'server_dyn_res': resval[0]}
+
+        self.scheduler.set_sched_config(a)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
```

Comment:
Why do you need to stop and start the scheduling cycle?

Reply:
The server dynamic resource script gets executed only when a new scheduling cycle kicks in . That's why.